### ISSUE TITLE
matplotlib figure unresponsive

### ISSUE THREAD

**Issue by rcbauck** [ipython/ipython#13526]
My first time submitting an issue, so sorry in advance for my ignorance, and for my lengthy description below...

I am launching IPython as shown below, and simply opening up two pyplot figures. If I close the first figure, the second ( and also any other additional figures that I may have opened before closing the first one ) become unresponsive. The IPython prompt is still alive and can be used, but any figures being displayed are "hung". If I try to close the figure(s), I get the Windows message that python is not responding, Close the program?

I can get the unclosed figure(s) to become responsive again if I simply re-create the original first figure, plt.figure(1) for example, or even do plt.figure(3).

C:\PythonProjects>ipython --matplotlib
Python 3.10.1 (tags/v3.10.1:2cd268a, Dec  6 2021, 19:10:37) [MSC v.1929 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.0.1 -- An enhanced Interactive Python. Type '?' for help.
Using matplotlib backend: TkAgg

Running C:/Users/Randy/.ipython/profile_default/startup/mystartup.py
mystartup.ipy finished

In [1]: from matplotlib import pyplot as plt

In [2]: plt.isinteractive()
Out[2]: True

In [3]: plt.figure(1)
Out[3]: <Figure size 640x480 with 0 Axes>

In [4]: plt.figure(2)
Out[4]: <Figure size 640x480 with 0 Axes>

In [5]: plt.close(1)

After this "close()", figure(2) is unresponsive..

This issue only happens when using IPython. The exact same sequence of commands works fine in the standard Python interactive terminal.

Another strange clue is that if I run IPython and I use the VS Code debugger and place a breakpoint anywhere after the "self._run_cell()" call in \core\interactiveshell.py module ( code snippet shown below ), and then do the import of pyplot, the matplotlib TkAgg backend is activated and automatically put in interactive mode, and the issue doesn't happen. NOTE, a Python breakpoint() statement does not make the issue go away, but a VS Code debugger breakpoint does.

        result = None
        try:
            result = self._run_cell(
                raw_cell, store_history, silent, shell_futures)
---> breakpoint anywhere after this eliminates the issue <-----
        finally:
            self.events.trigger('post_execute')
            if not silent:
                self.events.trigger('post_run_cell', result)
        
        return result

I used an older version of IPython with an older version of Python2 for over 10 years and didn't see this behavior.
I also tried getting a "7" version of IPython and it has the same issue as the "8" version.

Not a big show stopper, just appears to be some bug that makes the first figure created somehow critical / magical to the operation of the subsequent figures.

**Comment by rcbauck** [comment]
Thanks for the quick response and suggestion. I tried the following, did not change / fix the issue. 

C:\PythonProjects>ipython
Python 3.10.1 (tags/v3.10.1:2cd268a, Dec  6 2021, 19:10:37) [MSC v.1929 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.0.1 -- An enhanced Interactive Python. Type '?' for help.

Running C:/Users/Randy/.ipython/profile_default/startup/mystartup.py
mystartup.ipy finished

In [1]: %gui tk

In [2]: %matplotlib
Using matplotlib backend: TkAgg

In [3]: from matplotlib import pyplot as plt

In [4]: plt.figure(1)
Out[4]: <Figure size 640x480 with 0 Axes>

In [5]: plt.figure(2)
Out[5]: <Figure size 640x480 with 0 Axes>

In [6]: plt.close(1)

I'll keep poking around this issue a little, but I am very ignorant about the inner workings of IPython and event loops, so I doubt I will get anywhere.

I can work around this easily by always just creating a "dummy" first figure and never closing it. If I do that, all subsequent figures work as expected and can be opened, closed, interacted with at any time in any sequence as far as I can tell.

**Comment by rcbauck** [comment]
Just curious, did you ( or anyone ) try that sequence of commands on non-Windows OS system to verify the issue only exists on Windows?

**Comment by MrMino** [comment]
Does this issue affect you on Python 3.8?

Unfortunately I don't have a Windows PC on hand either, cannot reproduce this.

**Comment by rcbauck** [comment]
@MrMino I have replicated this on 3.10 and 3.7. Let me know if you think there's some reason to believe 3.7 has the problem, 3.8 doesn't, and then 3.10 does. I can always get 3.8 and try it.

**Comment by MrMino** [comment]
No, 3.7 was ok, it's just that 3.10 is relatively new, and I wanted that info on the offchance that it's related to the interpreter version. Seems it's not.

**Comment by rcbauck** [comment]
@MrMino, just correcting my last statement...after reviewing my earlier comments, I actually tried an earlier version of IPython, a "7" version, and it had the same issue, I haven't tried an earlier version of Python. I will try that.

But, just to be clear, this issue only happens when using IPython as the interpreter, with a Python 3.10 interpreter, there is no issue. Anyway, I can try IPython running on top of a Python 3.7 version and see what happens.

**Comment by rcbauck** [comment]
@MrMino, I just installed Python 3.7 and installed IPython ( 7.32.0 ) and matplot lib into that version and ran the same test. The bad behavior is the same...

In [2]: import matplotlib.pyplot as plt

In [3]: plt.figure(1)
Out[3]: <Figure size 640x480 with 0 Axes>

In [4]: plt.figure(2)
Out[4]: <Figure size 640x480 with 0 Axes>

In [5]: plt.close(1)

Figure 2 is now unresponsive / hung.

Recreating Figure 1...

In [6]: plt.figure(1)
Out[6]: <Figure size 640x480 with 0 Axes>

Figure 1 and Figure 2 are now both responsive, work normally.

Good idea to try that, looks like the issue exists at least back to Python 3.7 and IPython 7.32.0.

**Comment by dfujim** [comment]
I have a similar issue. Steps to reproduce: 

```python
ipython --matplotlib

Python 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.
Using matplotlib backend: TkAgg

In [1]: plt.figure()
Out[1]: <Figure size 640x480 with 0 Axes>

In [2]: plt.figure()
Out[2]: <Figure size 640x480 with 0 Axes>

In [3]: plt.close(1)
```

Now figure 2 is unresponsive. If I instead close figure 2, everything is ok. 

Some version numbers: 

```
Python                  3.10.6
ipython                 8.5.0
matplotlib              3.6.1
matplotlib-inline       0.1.6
tk.TkVersion            8.6
```

Operating system: Ubuntu 22.04.1 LTS
